src/MultiOwnable.sol

```diff
@@ -1,124 +1,217 @@
+    ///
+    /// @param account The account to check identified by its address or passkey.
+    ///
+    /// @return `true` if the account is an owner, else `false`.
+    function isOwnerBytes(bytes memory account) public view virtual returns (bool) {
+        return _getMultiOwnableStorage().isOwner[account];
+    }
+
+    /// @notice Returns the owner bytes at the given `index`.
+    ///
+    /// @param index The index to lookup.
+    ///
+    /// @return The owner bytes (empty if no owner is registered at this `index`).
     function ownerAtIndex(uint256 index) public view virtual returns (bytes memory) {
         return _getMultiOwnableStorage().ownerAtIndex[index];
     }
 
+    /// @notice Returns the next index that will be used to add a new owner.
+    ///
+    /// @return The next index that will be used to add a new owner.
     function nextOwnerIndex() public view virtual returns (uint256) {
         return _getMultiOwnableStorage().nextOwnerIndex;
     }
 
+    /// @notice Initialize the owners of this contract.
+    ///
+    /// @dev Intended to be called when the smart account is first deployed.
+    /// @dev Reverts if a provided owner is neither 64 bytes long (for passkey) nor a valid address.
+    ///
+    /// @param owners The intiial list of owners to register.
     function _initializeOwners(bytes[] memory owners) internal virtual {
-        for (uint256 i = 0; i < owners.length; i++) {
+        for (uint256 i; i < owners.length; i++) {
             if (owners[i].length != 32 && owners[i].length != 64) {
                 revert InvalidOwnerBytesLength(owners[i]);
             }
+
             if (owners[i].length == 32 && uint256(bytes32(owners[i])) > type(uint160).max) {
                 revert InvalidEthereumAddressOwner(owners[i]);
             }
-            _addOwnerAtIndex(owners[i], _getMultiOwnableStorage().nextOwnerIndex++);
+
+            _addOwnerAtIndex({owner: owners[i], index: _getMultiOwnableStorage().nextOwnerIndex++});
         }
     }
 
+    /// @notice Convenience function used to add the first 255 owners.
+    ///
+    /// @param owner The owner raw bytes to add.
     function _addOwner(bytes memory owner) internal virtual {
-        _addOwnerAtIndex(owner, _getMultiOwnableStorage().nextOwnerIndex++);
+        _addOwnerAtIndex({owner: owner, index: _getMultiOwnableStorage().nextOwnerIndex++});
     }
 
+    /// @notice Adds an owner at the given `index`.
+    ///
+    /// @dev Reverts if `owner` is already registered as an owner.
+    ///
+    /// @param owner The owner raw bytes to register.
+    /// @param index The index to write to.
     function _addOwnerAtIndex(bytes memory owner, uint256 index) internal virtual {
         if (isOwnerBytes(owner)) revert AlreadyOwner(owner);
 
         _getMultiOwnableStorage().isOwner[owner] = true;
         _getMultiOwnableStorage().ownerAtIndex[index] = owner;
 
-        emit AddOwner(index, owner);
+        emit AddOwner({index: index, owner: owner});
```

Comment:
what's the thought here? We don't use type args elsewhere?

Reply:
No we don't but I think we should use them more. It helps for readability and increase confidence about writing sound contracts (avoid swapping params).

Reply:
I actually think they are more cluttering than helpful. We should lean on good variable names to avoid swapping params

Reply:
yeah I find it redundant here and we should be consistent with our events

Reply:
I understand your point and agree that it might be redundant, so I will remove them.

However, from my experience, using named parameters has the great benefit of ensuring, without any context about what the function does, nor about its parameter names and types, that the call is most likely correct. When conducting code reviews, I find it much easier to read, as I don't have to hop between the function definition and the function call to make sure the developer wrote what they really intended. You don't have to be a skilled auditor to notice that `transfer({to: from, from: to, amount: amount})` is incorrect, which I believe is important if we want to provide secure and foolproof open-source code.

Reply:
Yes thanks @xenoliss, let's file these away for our style guide discussion, after which we can commit to using the same approach in all places